expression/builtin_arithmetic.go

```diff
@@ -634,8 +659,11 @@ func (s *builtinArithmeticIntDivideDecimalSig) evalInt(row []types.Datum) (int64
 
 	c := &types.MyDecimal{}
 	err = types.DecimalDiv(a, b, c, types.DivFracIncr)
+	if err == types.ErrDivByZero {
+		return 0, true, errors.Trace(handleDivisionByZeroError(s.ctx))
+	}
 	if err != nil {
-		return 0, err == types.ErrDivByZero, errors.Trace(err)
+		return 0, false, errors.Trace(err)
```

Comment:
`return 0, true, errors.Trace(err)`？

Reply:
I don't think other exceptions need to return `NULL`.

Reply:
why ?

Reply:
Sorry, my fault. You are right. Forget it.